src/Domain/DonationService.php

```diff
@@ -488,6 +489,9 @@ function () use ($donation) {
     /**
      * Sets donation to cancelled in matchbot db, releases match funds, cancels payment in stripe, and updates
      * salesforce.
+     *
+     * Call this from inside a transaction and with a locked donation to avoid double releasing funds associated with
```

Comment:
Nice, makes sense. Took me a moment to see why the cancel call via `MatchBot\Application\Actions\Donations\Update` is safe despite not using `releaseMatchFundsInTransaction()` but I think it is. Possibly worth a comment in `Update` to emphasise how they relate and the importance of not removing the `beginTransaction()` stuff?

Reply:
Right, that's safe because the donation is locked at https://github.com/thebiggive/matchbot/blob/b9ff400bc78ad48a23f9865cbc1bfff3863d0cd4/src/Application/Actions/Donations/Update.php#L118 I think.

I'll add a comment for now. I did also wonder about if we could have a `LockedDonation` type somehow. Not sure about that.

Reply:
Done in c30771bdc4684fe530ac